src/core/processing/qgsprocessingparametertininputlayers.h

```diff
@@ -0,0 +1,112 @@
+/***************************************************************************
+  qgsprocessingparametertininputlayers.h
+  ---------------------
+  Date                 : August 2020
+  Copyright            : (C) 2020 by Vincent Cloarec
+  Email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPROCESSINGPARAMETERTININPUTLAYERS_H
+#define QGSPROCESSINGPARAMETERTININPUTLAYERS_H
+
+#include "qgsprocessingparameters.h"
+#include "qgsprocessingparametertype.h"
+
+/**
+ * A parameter for processing algorithms that need a list of input vector layers to construct a TIN
+ * A valid value for this parameter is a list (QVariantList), where each item is a map (QVariantMap) in this form:
+ * {
+ *   'Id':  string hat represents the unique Id of the vector layer,
+ *   'Type': how the vector layer is used : as vertices or as break lines
+ *   'AttributeIndex' : the index of the attribute of the vector layer used to defined the Z value of vertices,
+ *    if -1, the Z coordinates of features are used
+ * }
+ *
+ * \ingroup core
+ * \since QGIS 3.16
+ */
+class CORE_EXPORT QgsProcessingParameterTinInputLayers: public QgsProcessingParameterDefinition
+{
+  public:
+
+    //! Defines the type of input layer
+    enum Type
+    {
+      Vertices, //! Input that adds only vertices
+      BreakLines //! Input that adds vertices and break lines
```

Comment:
To keep compatibility with the existing TIN interpolation algorithm (and allow for that algorithm to be ported to the new parameter type in future), can you add "StructureLines" as value 1 here?